### PR TITLE
Created Entity and Category processing lambda

### PR DESCRIPTION
**Pull request by Cortey**
Added Entity and Category processing functionality

Changes:
- Entity and entity sentiment function
- Category function
- Subscriptions for new functions
- Added event generation in text-font function

### CODE REVIEW

File: resources/functions/labels-text.yaml [hunk 1]
```diff
@@ -89,15 +90,53 @@ spec:
...
+          .catch(function (error) {
+            console.error(error);
+          });
+        }
+
         try{
           const imgID = event.data.slice(1, -1);
-
+          console.log("Extension" + event.extensions.request.body);
+          console.log("Eventdata" + event.data);
```
Comment (mwisnia97) [inline resources/functions/labels-text.yaml:122]: Is it necessary?

File: resources/functions/text-category.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          content: text,
+          type: 'PLAIN_TEXT',
+        };
+
+        const [classification] = await client.classifyText({document});
+        const results = classification.categories.map(category => ({
+          name: category.name,
+          confidence: category.confidence,
+        }));
+        return results;
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:69]: Add clg with information that we get categories.

File: resources/functions/text-category.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      main: async function (event, context) {
+        async function getText(id) {
+          console.log("Getting font from DB");
+          const fetch = require('node-fetch');
+          const URL = JSON.parse(process.env.API_CONFIG).API_URL + '/' + id;
+          const response = await fetch(URL);
+          const data = await response.json();
+          const content = data.gcp.map(JSON.parse);
+          const font = content?.find(obj => Object.keys(obj).includes('font'))?.font || [];
+          console.log("Text loaded");
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:44]: Getting font and then text loaded. Please name it in one convention.

File: resources/functions/text-entity.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      main: async function (event, context) {
+        async function getText(id) {
+          console.log("Getting font from DB");
+          const fetch = require('node-fetch');
+          const URL = JSON.parse(process.env.API_CONFIG).API_URL + '/' + id;
+          const response = await fetch(URL);
+          const data = await response.json();
+          const content = data.gcp.map(JSON.parse);
+          const font = content?.find(obj => Object.keys(obj).includes('font'))?.font || [];
+          console.log("Text loaded");
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:44]: The same here. One convention please.

File: resources/functions/text-category.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        const fetch = require('node-fetch');
+        const URL = JSON.parse(process.env.API_CONFIG).API_URL + '/' + id;
+        const response = await fetch(URL,{
+          method:'PUT',
+          body: data,
+          headers: {
+            'Content-type': 'application/json',
+          },
+        });
+        await console.log("Entity details put to DB with status: " + response.status);
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:84]: It's not entity. Text category put to DB

File: resources/functions/text-entity.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          type: 'PLAIN_TEXT',
+        };
+        const [result] = await client.analyzeEntitySentiment({document});
+        const entities = result.entities;
+        const results = entities.map(entity => ({
+        name: entity.name,
+        type: entity.type,
+        sentiment: entity.sentiment.score,
+        })
+        );
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:70]: Add information that entity is loaded.

File: resources/functions/text-entity.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+          content: text,
+          type: 'PLAIN_TEXT',
+        };
+        const [result] = await client.analyzeEntitySentiment({document});
+        const entities = result.entities;
+        const results = entities.map(entity => ({
+        name: entity.name,
+        type: entity.type,
+        sentiment: entity.sentiment.score,
+        })
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:69]: Tabulator it please.

File: resources/subscriptions/text-entity-sub.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    filters:
+      - eventSource:
+          property: source
+          type: exact
+          value: ""
+        eventType:
+          property: type
+          type: exact
+          value: sap.kyma.custom.showcase.textdetails.processing.v1
+  sink: http://labels-text.default.svc.cluster.local
```
Comment (mwisnia97) [inline resources/subscriptions/text-entity-sub.yaml:16]: Sink is wrong, text-entity is correct I think

File: resources/subscriptions/text-category.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    filters:
+      - eventSource:
+          property: source
+          type: exact
+          value: ""
+        eventType:
+          property: type
+          type: exact
+          value: sap.kyma.custom.showcase.textdetails.processing.v1
+  sink: http://labels-text.default.svc.cluster.local
```
Comment (mwisnia97) [inline resources/subscriptions/text-category.yaml:16]: Sink is wrong, text-category is correct I think

File: resources/functions/labels-text.yaml [hunk 10]
```diff
@@ -66,7 +67,7 @@ spec:
           const resultTexts = result.textAnnotations;
           const fontDetails = resultTexts.map(text => text.description);
           if (fontDetails.length === 0)
-            return null;
+            return null
```
Comment (mwisnia97) [inline resources/functions/labels-text.yaml:70]: ??? Bring ; free again

**Review by mwisnia97 — CHANGES_REQUESTED** [review]

File: resources/functions/text-category.yaml (hunk 2 above)
Comment (Cortey) [reply resources/functions/text-category.yaml:69]: Line 49

Comment (mwisnia97) [reply resources/functions/text-category.yaml:69]: No, it's first line informing that function start working, add clg that function finished task and everything is fine.

File: resources/functions/text-entity.yaml [hunk 11]
```diff
@@ -63,11 +63,12 @@ spec:
         const [result] = await client.analyzeEntitySentiment({document});
         const entities = result.entities;
         const results = entities.map(entity => ({
-        name: entity.name,
-        type: entity.type,
+          name: entity.name,
+          type: entity.type,
         sentiment: entity.sentiment.score,
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:68]: One more tabulator

File: resources/functions/text-entity.yaml [hunk 12]
```diff
@@ -63,11 +63,12 @@ spec:
...
         })
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:69]: Here too tabulator

**Review by mwisnia97 — CHANGES_REQUESTED** [review]
Cosmetic changes

**Review by mwisnia97 — CHANGES_REQUESTED** [review]
line 90-100 has one unnecessary tabulator.

File: resources/functions/text-category.yaml [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+          // Needed to remove unwanted apostrophes from our ID
+          const imgID = event.data.slice(3, -3);
+          console.log("Processing image: " + imgID);
+
+          const text = await getText(imgID);
+          const textCategory = await getTextCategory(text);
+          await putDetailsToDB(imgID, textCategory);
+          }catch(err){
+            console.error(err);
+            return null;
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:98]: Here too, one to much tabulator

**Review by mwisnia97 — CHANGES_REQUESTED** [review]
Clean code in this one place and LGTM 👋

File: resources/functions/text-category.yaml [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        const client = new language.LanguageServiceClient(options);
+        const document = {
+          content: text,
+          type: 'PLAIN_TEXT',
+        };
+        const [classification] = await client.classifyText({document});
+        const results = classification.categories.map(category => ({
+          name: category.name,
+          confidence: category.confidence,
+        }));
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:67]: image
Idk why, but it is a very very bad response to DB.

File: resources/functions/text-entity.yaml [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+        const document = {
+          content: text,
+          type: 'PLAIN_TEXT',
+        };
+        const [result] = await client.analyzeEntitySentiment({document});
+        const entities = result.entities;
+        const results = entities.map(entity => ({
+            name: entity.name,
+            type: entity.type,
+            sentiment: entity.sentiment.score,
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:68]: image
Idk why, but it is a very very bad response to DB.

**Review by mwisnia97 — CHANGES_REQUESTED** [review]

File: resources/functions/text-category.yaml [hunk 16]
```diff
@@ -91,7 +94,12 @@ specs:
         console.log("Processing image: " + imgID);
         const text = await getText(imgID);
         const textCategory = await getTextCategory(text);
-        await putDetailsToDB(imgID, textCategory);
+        if(textCategory){
+          await putDetailsToDB(imgID, textCategory);
+        }
+        else{
+          console.log("Category is empty");
```
Comment (mwisnia97) [inline resources/functions/text-category.yaml:101]: Category array is empty

File: resources/functions/text-entity.yaml [hunk 17]
```diff
@@ -93,7 +95,12 @@ spec:
         console.log("Processing image: " + imgID);
         const text = await getText(imgID);
         const textEntities = await getTextEntities(text);
-        await putDetailsToDB(imgID, textEntities);
+        if(textEntities){
+          await putDetailsToDB(imgID, textEntities);
+        }
+        else{
+          console.log("Text Entities empty")
```
Comment (mwisnia97) [inline resources/functions/text-entity.yaml:102]: Text entities array is empty

**Review by mwisnia97 — CHANGES_REQUESTED** [review]

**Review by mwisnia97 — APPROVED** [review]

**Pull request merged**